charts/pact-broker/templates/serviceaccount.yaml

```diff
@@ -1,19 +1,27 @@
 {{- if .Values.serviceAccount.create }}
+{{- with .Values.serviceAccount }}
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: {{ .Values.serviceAccount.name }}
-  namespace: {{ .Release.Namespace }}
+  name: {{ .name }}
```

Comment:
Not sure this is correct? We definitely want to keep the service account name, it's the same with the labels and annotations below.

Reply:
This is keeping the serviceName. Same as in the other PR. 

when using `with` like I do here: https://github.com/pact-foundation/pact-broker-chart/blob/ab1ebe6d2a463a0f01c648c55f81966c62779af0/charts/pact-broker/templates/serviceaccount.yaml#L2 it allows you to enter that context and reference everything from that key level. This is outlined: https://helm.sh/docs/chart_template_guide/control_structures/#modifying-scope-using-with but yeah.

Reply:
Ah, apologies, it was late at night, I didn't see the `with`. Although I don't see it that often in public Helm Charts if I'm honest, maybe because it is one of those features that tend to throw people off a bit instead of helping them cut code down, either way we'll see if it causes problems with readability for future contributors. If not, I'm happy with it staying in.